src/app/shared/services/auth/auth.service.ts

```diff
@@ -53,12 +66,20 @@ export class AuthService extends AsyncServiceBase {
     }
   }
 
-  private async checkProfileRestore(authUser?: IAuthUser) {
-    if (!authUser) return;
-    const existingUser = this.localStorageService.getProtected("AUTH_USER_ID");
-    if (existingUser) return;
-    // no existingUser user, should check if authUser exists on server and prompt restore
-    // TODO - handle
+  private async checkForUserRestore(authUser: IAuthUser) {
+    // List all server entries with the same auth_id (multiple devices)
+    // TODO - get type-safe return types using openapi http client
+    const authEntries = await firstValueFrom(
+      this.http
+        .get(`/auth_users/${authUser.uid}`, { responseType: "json" })
+        .pipe(map((v) => (v as IServerUser[]) || []))
+    );
+
+    const currentUserId = this.localStorageService.getProtected("APP_USER_ID");
+    const restoreProfiles = authEntries
+      .filter((v) => v.app_user_id !== currentUserId)
```

Comment:
I can see how there might be a use case for restoring from the same ID (e.g. after factory resetting a single device). But it makes sense to focus on the requested use case for now